tactics/Goalie.py

```diff
@@ -121,28 +189,30 @@ def execute_peace(self):
 	def execute_clear(self):
 		print("Clear kar raha hoon")
 		target = Vector2D(4000,0)
-		state = shared.get('state')
-		self.gtB = GoToBall.GoToBall(BOT_BALL_THRESH)
-		self.gtB.add_kub(self.kub)
-		self.gtB.add_theta(theta=normalize_angle(atan2(target.y - state.ballPos.y,target.x - state.ballPos.x)))
-		self.gtB.spin()
-
-		#Will Use KickToPoint after making it perfect with kicking
-
-		# self.ktp = KickToPoint.KickToPoint(target)
-		# self.ktp.add_kub(self.kub)
-		# self.ktp.add_theta(theta=normalize_angle(atan2(target.y - state.ballPos.y,target.x - state.ballPos.x)))
-		# self.ktp.spin()
+		state = None
+		try:
+			state = getState(state)
+		except rospy.ServiceException, e:
+			print e
+		if state:
+			state = state.stateB
+		self.ktp = KickToPoint.KickToPoint(target)
+		self.ktp.add_kub(self.kub)
```

Comment:
Do you think you might need to update kub state before adding kub?

Reply:
If yes, then make sure you do it and update kub state everywhere, before adding it to kicktopoint.